user/views.py

```diff
@@ -1,3 +1,65 @@
-from django.shortcuts import render
+import jwt
+import json
+import requests
 
-# Create your views here.
+from django.views import View
+from django.http import JsonResponse
+
+from config.settings import (
+    SECRET_KEY,
+    ALGORITHM
+)
+
+from .models import Users
+from .utils import login_decorator
+
+KAKAO_API = 'https://kapi.kakao.com/v2/user/me'
+
+class SocialLoginView(View):
+    def post(self, request):
+        try:
+            kakao_token         = request.headers['Authorization']
+            req_kakao_profile   = requests.get(KAKAO_API,headers = {
+                'Authorization' : f'Bearer {kakao_token}'
+            })
+            profile         = req_kakao_profile.json()
+            kakao_email     = profile['kakao_account']['email']
+            kakao_id        = profile['properties']['nickname']
+            kakao_picture   = profile['properties']['profile_image']
+
+            if Users.objects.filter(email = kakao_email).exists():
+                token = jwt.encode({'email' : kakao_email}, SECRET_KEY, algorithm=ALGORITHM)
+                token = token.decode('utf-8')
+                return JsonResponse({
+                    'access_token'      : token,
+                    'nickname'          : kakao_id,
+                    'profile_image'     : kakao_picture,
+                    'email'             : kakao_email
+                }, status=200)
+
+            Users(
+                email           = kakao_email,
+                kakao_id        = kakao_id,
+                picture         = kakao_picture,
+            ).save()
+
+            token = jwt.encode({'email' : kakao_email}, SECRET_KEY, algorithm=ALGORITHM)
+            token = token.decode('utf-8')
```

Comment:
token에 다시 재할당하지 않고 바로 적용하셔도 될 것 같습니다. :)
```suggestion
            token = jwt.encode({'email' : kakao_email}, SECRET_KEY, algorithm=ALGORITHM).decode('utf-8')
```